pkg/splunk/enterprise/configuration_test.go

```diff
@@ -236,3 +236,229 @@ func TestSetVolumeDefault(t *testing.T) {
 		}
 	}
 }
+
+func TestSmartstoreApplyIndexerClusterFailsOnInvalidSmartStoreConfi(t *testing.T) {
+	cr := enterprisev1.IndexerCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "standalone",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.IndexerClusterSpec{
+			Replicas: 1,
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", S3Endpoint: ""},
+				},
+
+				IndexList: []enterprisev1.IndexSpec{
+					{Name: "salesdata1", RemoteLocation: "testbucket-rs-london/$_index_name"},
+					{Name: "salesdata2", RemoteLocation: "testbucket-rs-london/$_index_name"},
+					{Name: "salesdata3", RemoteLocation: "testbucket-rs-london/$_index_name"},
+				},
+			},
+		},
+	}
+
+	var client splcommon.ControllerClient
+
+	_, err := ApplyIndexerCluster(client, &cr)
+	if err == nil {
+		t.Errorf("ApplyIndexerCluster should fail on invalid smartstore config")
+	}
+}
+
+func TestSmartstoreApplyStandaloneFailsOnInvalidSmartStoreConfi(t *testing.T) {
+	cr := enterprisev1.Standalone{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "standalone",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.StandaloneSpec{
+			Replicas: 1,
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", S3Endpoint: ""},
+				},
+
+				IndexList: []enterprisev1.IndexSpec{
+					{Name: "salesdata1", RemoteLocation: "testbucket-rs-london/$_index_name"},
+					{Name: "salesdata2", RemoteLocation: "testbucket-rs-london/$_index_name"},
+					{Name: "salesdata3", RemoteLocation: "testbucket-rs-london/$_index_name"},
+				},
+			},
+		},
+	}
+
+	var client splcommon.ControllerClient
+
+	_, err := ApplyStandalone(client, &cr)
+	if err == nil {
+		t.Errorf("ApplyStandalone should fail on invalid smartstore config")
+	}
+}
+
+func TestSmartStoreConfigDoNotFailsOnSitesWithCM(t *testing.T) {
+	cr := enterprisev1.IndexerCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "idx_CM_site",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.IndexerClusterSpec{
+			Replicas: 3,
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", S3Endpoint: "https://s3-eu-west-2.amazonaws.com"},
+				},
+
+				IndexList: []enterprisev1.IndexSpec{
+					{Name: "salesdata1", RemoteLocation: "testbucket-rs-london/$_index_name"},
+					{Name: "salesdata2", RemoteLocation: "testbucket-rs-london/$_index_name"},
+					{Name: "salesdata3", RemoteLocation: "testbucket-rs-london/$_index_name"},
+				},
+			},
+		},
+	}
+
+	err := validateIndexerClusterSpec(&cr)
+
+	if err != nil {
+		t.Errorf("Smartstore configuration is Invalid: %v", err)
+	}
+}
+
+func TestSmartStoreConfigFailsOnSitesWithNoCM(t *testing.T) {
+	cr := enterprisev1.IndexerCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "idexer_site_1",
```

Comment:
ditto:same as above

Reply:
same as above: All these changes are assuming the multi site approach that is what we have today.

Reply:
I think you meant multi-part?